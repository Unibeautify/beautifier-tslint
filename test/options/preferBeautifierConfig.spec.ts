import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "../../src";
import * as fs from "fs";
import * as path from "path";
import { raw } from "../utils";
const filePath: string = path.resolve(
  __dirname,
  "../fixtures/preferBeautifierConfig/test2.ts"
);
test(`should place a comma at the end of the typescript object`, () => {
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  const text: string = fs
    .readFileSync(
      path.resolve(__dirname, `../fixtures/preferBeautifierConfig/test2.ts`)
    )
    .toString();
  return unibeautify
    .beautify({
      filePath,
      languageName: "TypeScript",
      options: {
        TypeScript: {
          TSLint: {
            prefer_beautifier_config: true,
          },
        } as any,
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
