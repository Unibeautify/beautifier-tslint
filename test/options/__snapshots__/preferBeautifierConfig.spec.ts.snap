// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should place a comma at the end of the typescript object 1`] = `
Object {
  Symbol(raw): "export const beautifier = {
  TypeScript: {
    end_with_comma: true,
  },
};
",
}
`;
